.travis/travis.py

```
@@ -16,95 +49,198 @@ def test_diffs(diffs):
+                create_dockerlibrary = import_create_dockerlibrary(
+                    hub_repo_dir)
+                create_dockerlibrary.main((
+                    '--manifest', manifest,
+                    '--output', output))
+
+                # Commit changes to Docker Library
+                repo.git.add(all=True)
+                message = "Updating Docker Library\n" + \
+                    "This is an automated CI commit"
+                repo.git.commit(m=message, author=GIT_AUTHOR)
+
+                # Create new branch from current head
+                new_branch = repo.create_head(pr_branch_name)
+
+                try:
+                    # Check if branch already exists
+                    g_branch = g_origin_repo.get_branch(branch=pr_branch_name)
+                    try:
+                        repo.git.push(
+                            'upstream_pr', pr_branch_name + ':' + pr_branch_name, force=True)
+                    except Exception as inst:
+                        inst.stderr = None
+                        raise ValueError(
+                            ("Force push to branch:{branch} failed! "
+                             "Stderr ommited to protect secrits.").format(branch=pr_branch_name))
+
+                except github.GithubException as exception:
+                    # Continue if branch does not already exist
+                    if exception.data['message'] == "Branch not found":
+                        try:
+                            repo.git.push(
+                                'upstream_pr', pr_branch_name + ':' + pr_branch_name, u=True)
+                        except Exception as inst:
+                            inst.stderr = None
+                            raise ValueError(
+                                ("Set-upstream push to branch:{branch} failed! "
+                                 "Stderr ommited to protect secrits.").format(branch=pr_branch_name))
```

what happens if the error message is NOT "Branch not found", I think it should raise the exception instead of continuing isn't it?

---

Fixed